proto/iprules.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package iprules;
+
+import "proto/context.proto";
+
+message IPRule {
+  string ip_rule_id = 1;
+
+  // Allowed IP range, in CIDR notation (e.g. 1.2.3.0/24)
+  string cidr = 2;
+
+  string description = 3;
+}
+
+message AddRuleRequest {
+  context.RequestContext request_context = 1;
+
+  IPRule rule = 2;
+}
+
+message AddRuleResponse {
+  context.ResponseContext response_context = 1;
+
+  IPRule rule = 2;
+}
```

Comment:
This is pretty close to Update -- could we get away with just having Update?

Reply:
Having Add and Update as separate operations avoids the need to have condition DB logic in the implementation for INSERT vs UPDATE. I prefer having separate operations.